### PR TITLE
DS-3195 Add check for syntactic names

### PR DESCRIPTION
**Pull request by jrwishart**
The call to base::alias used for RIA and protecting against aliased is
fragile and cannot handle non-syntactic names. This commit inspects the
formula for non-syntactic variable names and updates the formula and
estimation data before running the alias checks

### CODE REVIEW

File: R/data.R [hunk 1]
```diff
@@ -331,6 +345,24 @@ validateDataForRIA <- function(input.formula, estimation.data, outcome.name, sho
...
+syntatic.name.patt <- "^((([[:alpha:]]|[.][._[:alpha:]])[._[:alnum:]]*)|[.])$"
+dataset.reference.patt <- "^[[:print:]]*[$](Variables|Questions)[$]"
+
+makeVariableNamesValid <- function(vars.to.relabel, remove.patt = NULL)
+{
+    new.var.names <- names(vars.to.relabel)
+    if (!is.null(remove.patt))
+        new.var.names <- sub(remove.patt, "", new.var.names)
+    new.var.names <- make.names(new.var.names)
+    if (any(duplicated(new.var.names)))
```
Comment (mwmclean) [inline R/data.R]: 1. `make.names` has an argument `unique`
2. (irrelevant here given 1.) there is a more efficient function for `any(duplicated(...))` called `anyDuplicated`

File: R/data.R [hunk 2]
```diff
@@ -313,9 +313,23 @@ validateDataForRIA <- function(input.formula, estimation.data, outcome.name, sho
...
+        estimation.data <- relabelled.outputs$data
+        outcome.name <- relabelled.outputs$outcome.name
+    }
+    # Check names are syntatic for formula and data, if non-syntatic, make them that way.
+    if (any(non.syntatic.vars <- checkFormulaForValidNames(input.formula, data = estimation.data,
+                                                           patt = syntatic.name.patt,
+                                                           negate = TRUE)))
+    {
+        new.var.names <- makeVariableNamesValid(non.syntatic.vars)
+        relabelled.outputs <- relabelFormulaAndData(new.var.names, input.formula, estimation.data,
```
Comment (mwmclean) [inline R/data.R:331]: `relabelFormulaAndData()` doesn't use its `patt` argument.

Comment (jrwishart) [reply R/data.R:331]: Thanks, that function got refactored and the arg no longer used. I have removed that arg and updated the documentation of that function to explain the new use (it still had the documentation explaining the old use). Updated in https://github.com/Displayr/flipRegression/pull/36/commits/6415d1042f282894823953a6366c4ed4f56d7398

File: R/data.R (hunk 1 above)
Comment (jrwishart) [reply R/data.R]: Updated in https://github.com/Displayr/flipRegression/pull/36/commits/6415d1042f282894823953a6366c4ed4f56d7398, thanks for picking that up. Removed the call to `duplicated` and updated the `make.names` call to use the `unique` argument.

**Review by mwmclean — APPROVED** [review]

**Pull request merged**